CAVEATS.md

```diff
@@ -0,0 +1,232 @@
+<!--
+    Copyright 2014, General Dynamics C4 Systems
+
+    SPDX-License-Identifier: GPL-2.0-only
+-->
+
+# Known caveats in the seL4 API and implementation
+
+## Implementation Correctness
+
+The following seL4 architectures have platforms with a C-level functional
+correctness proof. Proof support for further platforms within these
+architectures is on the roadmap and expected in 2025.
+
+- AArch32: Armv7-a with and without hypervisor extensions, no SMMU, with
+  fast path
+  - Platforms (non-hyp): `sabre` (no FPU), `imx8mm-evk` (with FPU)
+  - Platforms (hyp, no FPU): `tk1`, `exynos5`
+- AArch64: Armv8-a with hypervisor extensions only, no SMMU, with fast path
+  - Platforms: `tx2`
+- RISC-V: 64-bit only, no fast path
+  - Platforms: `hifive`
+- x64: without VT-x and VT-d, no fast path
+  - Platforms: `pc99`
+
+This proof covers the functional behaviour of the C code of the kernel. It does
+not cover machine code, compiler, linker, boot code, cache and TLB management.
+Compiler and linker can be removed from this list by additionally running the
+binary verification tool chain for seL4 for AArch32 or RISC-V. The proof shows
+that the seL4 C code implements the formal [abstract API specification][ASpec]
+of seL4 and is free from standard C implementation defects such as buffer
+overruns or NULL pointer dereferences.
```

Comment:
This makes it sound a bit like the binary verification proof shows that ASpec is refined by the C code. Can leave it, as "the proof" refers to "this proof" earlier, and if someone doesn't know binary verification is a proof, it'll probably be ok.

Reply:
Could say "The overall proof shows". Would that be clearer?

Reply:
Just to add another data point, I had the same thought about first reading the second half of this paragraph as still talking about binary verification.

Reply:
The "overall" might help clarify here, yes. Probably not worth re-engineering the whole paragraph.